src/main/java/org/folio/rest/transaction/EncumbranceHandler.java

```diff
@@ -0,0 +1,107 @@
+package org.folio.rest.transaction;
+
+import static org.folio.rest.impl.TransactionSummaryAPI.ORDER_TRANSACTION_SUMMARIES;
+import static org.folio.rest.persist.HelperUtils.getCriterionByFieldNameAndValueNotJsonb;
+import static org.folio.rest.persist.HelperUtils.handleFailure;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.resource.FinanceStorageTransactions;
+import org.folio.rest.persist.Tx;
+import org.folio.rest.persist.Criteria.Criterion;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+public class EncumbranceHandler extends AllOrNothingHandler {
+
+  private static final String TEMPORARY_ORDER_TRANSACTIONS = "temporary_order_transactions";
+
+  public EncumbranceHandler(Map<String, String> okapiHeaders, Context ctx, Handler<AsyncResult<Response>> asyncResultHandler) {
+    super(TEMPORARY_ORDER_TRANSACTIONS, ORDER_TRANSACTION_SUMMARIES, okapiHeaders, ctx, asyncResultHandler);
+  }
+
+  @Override
+  public void updateTransaction(Transaction transaction) {
+    verifyTransactionExistence(transaction.getId())
+      .compose(aVoid -> processAllOrNothing(transaction, this::updatePermanentTransactions))
+      .setHandler(result -> {
+        if (result.failed()) {
+          HttpStatusException cause = (HttpStatusException) result.cause();
+          switch (cause.getStatusCode()) {
+          case 400:
+            getAsyncResultHandler().handle(Future.succeededFuture(
+                FinanceStorageTransactions.PutFinanceStorageTransactionsByIdResponse.respond400WithTextPlain(cause.getPayload())));
+            break;
+          case 404:
+            getAsyncResultHandler().handle(Future.succeededFuture(
+                FinanceStorageTransactions.PutFinanceStorageTransactionsByIdResponse.respond404WithTextPlain(cause.getPayload())));
+            break;
+          default:
+            getAsyncResultHandler()
+              .handle(Future.succeededFuture(FinanceStorageTransactions.PutFinanceStorageTransactionsByIdResponse
+                .respond500WithTextPlain(Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase())));
+          }
+        } else {
+          log.debug("Preparing response to client");
+          getAsyncResultHandler()
+            .handle(Future.succeededFuture(FinanceStorageTransactions.PutFinanceStorageTransactionsByIdResponse.respond204()));
+        }
+      });
+  }
+
+  @Override
+  String getId(Transaction transaction) {
+    return transaction.getEncumbrance()
+      .getSourcePurchaseOrderId();
+  }
+
+  @Override
+  Criterion getCriterion(String value) {
+    return getCriterionByFieldNameAndValueNotJsonb("encumbrance_sourcepurchaseorderid", "=", value);
```

Comment:
shouldn't this be camel case? encumbrance_sourcePurchaseOrderId?

Reply:
It's OK. The field name is in lowercase 
![image](https://user-images.githubusercontent.com/45555481/69144560-98270b80-0adc-11ea-9633-827be70d488c.png)

Reply:
I can update this to improve readability.